### PR TITLE
add async-graphql-tide

### PR DESCRIPTION
**Pull request by vkill**
Hi sunli829 , please review it. thank you.

### CODE REVIEW

**Comment by vkill** [comment]
I will update doc and examples after this PR merges.

**Comment by sunli829** [comment]
Thank you for your contribution to `async-graphql`. I think it's done a pretty good job😁, but a few minor changes are needed, support for multipart requests for file upload. `Async-graphql` has a multipart parser, here is the implementation (https://github.com/sunli829/async-graphql/blob/0607c6da56da2a801120bdd4dc1aa9c42b921fea/src/http/into_query_builder.rs#L10). Create a `QueryBuilder` with a tuple that contains the `Option<ContentType>` and `AsyncRead`.

Here is an example:
https://github.com/sunli829/async-graphql/blob/0607c6da56da2a801120bdd4dc1aa9c42b921fea/async-graphql-warp/src/lib.rs#L82

`StreamBody` converts `Stream<Item=Bytes>` to `AsyncRead`, which you probably won't use because 'tide::Body' is `AsyncRead`.

File: async-graphql-tide/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "async-graphql-tide"
+version = "1.0.0"
+authors = ["sunli <scott_s829@163.com>"]
```
Comment (sunli829) [inline async-graphql-tide/Cargo.toml:4]: Change it to your name.😁

**Comment by vkill** [comment]
>  Create a `QueryBuilder` with a tuple that contains the `Option<ContentType>` and `AsyncRead`.

Please review again.

**Comment by sunli829** [comment]
It works fine, thanks again.😁

**Comment by sunli829** [comment]
你在rust编程语言的微信群里面吗？😁

**Comment by vkill** [comment]
> 你在rust编程语言的微信群里面吗？

没，才写 rust 一个多月，群是啥？

**Comment by sunli829** [comment]
你加我微信，sl162587，我拉你进去

**Comment by vkill** [comment]
已加

**Pull request merged**